x-pack/plugins/fleet/server/services/preconfiguration/outputs.ts

```diff
@@ -221,6 +338,7 @@ function isPreconfiguredOutputDifferentFromCurrent(
     isDifferent(existingOutput.config_yaml, preconfiguredOutput.config_yaml) ||
     isDifferent(existingOutput.proxy_id, preconfiguredOutput.proxy_id) ||
     isDifferent(existingOutput.allow_edit ?? [], preconfiguredOutput.allow_edit ?? []) ||
-    kafkaFieldsAreDifferent()
+    (await kafkaFieldsAreDifferent()) ||
+    (await logstashFieldsAreDifferent())
```

Comment:
Could we cover kafka and logstash output changes in unit tests?

Reply:
Thanks for that @juliaElastic - your suggestion made me realise the tests were not properly testing the comparison of hashes. I pushed a commit that addresses that.